extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -197,19 +207,180 @@ const buildExpandedViewOverlay = element => htmlFor(element)`
 
 /**
  * Updates embed's corresponding <style> element with embedData.
+ * @param {!Element} target
  * @param {!Element} embedStyleEl
  * @param {!EmbedDataDef} embedData
  */
-function updateEmbedStyleEl(embedStyleEl, embedData) {
+function updateEmbedStyleEl(target, embedStyleEl, embedData) {
   const embedId = embedData.id;
+  embedStyleEl.textContent = `[${EMBED_ID_ATTRIBUTE_NAME}="${embedId}"]
+  ${buildStringStyleFromEl(target, embedData)}`;
+}
+
+/**
+ * Builds a string containing the corresponding style depending on the
+ * element.
+ * @param {!Element} target
+ * @param {!EmbedDataDef} embedData
+ * @return {string}
+ */
+function buildStringStyleFromEl(target, embedData) {
```

Comment:
I really like how you extracted all the logic in simple and small functions :)) Any reason why you want them as separate functions instead of class methods?
If we're not planning on exporting them I think we should keep them as private class methods

Reply:
Thanks :) The functions have to be outside of the class scope since `prepareForAnimation` (the function that calls them) is a static function called outside of an object instance, so there is no access to the internal class methods when it is being called.

Reply:
Sounds good, thanks :)